src/cargo-kani/src/session.rs

```diff
@@ -0,0 +1,234 @@
+            cmd.stdout(std::process::Stdio::null());
+            cmd.stderr(std::process::Stdio::null());
+        }
+        if self.args.verbose || self.args.dry_run {
+            println!("{}", render_command(&cmd).to_string_lossy());
+            if self.args.dry_run {
+                // Short circuit
+                return Ok(());
+            }
+        }
+        let result = cmd
+            .status()
+            .context(format!("Failed to invoke {}", cmd.get_program().to_string_lossy()))?;
+        if !result.success() {
+            bail!("{} exited with status {}", cmd.get_program().to_string_lossy(), result);
+        }
+        Ok(())
+    }
+
+    /// Run a job, but only output (unless --quiet) if it fails, and fail if there's a problem.
+    pub fn run_suppress(&self, mut cmd: Command) -> Result<()> {
+        if self.args.quiet || self.args.debug || self.args.verbose || self.args.dry_run {
+            return self.run_terminal(cmd);
+        }
+        let result = cmd
+            .output()
+            .context(format!("Failed to invoke {}", cmd.get_program().to_string_lossy()))?;
+        if !result.status.success() {
+            // Don't suppress the output. There doesn't seem to be a way to easily get Command
+            // to give one output stream of both out/err with interleaving correct, it seems
+            // you'd have to resort to some lower-level interface.
+            let stdout = std::io::stdout();
+            let mut handle = stdout.lock();
+            handle.write_all(&result.stdout)?;
+            handle.write_all(&result.stderr)?;
+            bail!("{} exited with status {}", cmd.get_program().to_string_lossy(), result.status);
+        }
+        Ok(())
+    }
+
+    /// Run a job, redirect its output to a file, and allow the caller to decide what to do with failure.
+    pub fn run_redirect(&self, mut cmd: Command, stdout: &Path) -> Result<ExitStatus> {
+        if self.args.verbose || self.args.dry_run {
+            println!("{} > {}", render_command(&cmd).to_string_lossy(), stdout.display());
+            if self.args.dry_run {
+                // Short circuit. Difficult to mock an ExitStatus :(
+                return Ok(<ExitStatus as std::os::unix::prelude::ExitStatusExt>::from_raw(0));
+            }
+        }
+        let output_file = std::fs::File::create(&stdout)?;
+        cmd.stdout(output_file);
+
+        return cmd
+            .status()
+            .context(format!("Failed to invoke {}", cmd.get_program().to_string_lossy()));
+    }
+}
+
+impl InstallType {
+    pub fn new() -> Result<Self> {
+        let mut exe = std::env::current_exe()
+            .context("cargo-kani was unable to determine where its executable was located")?;
+        // Remove the executable name, so we're in the directory we care about
+        exe.pop();
+
+        // Case 1: We've checked out the development repo and we're built under `target/`
+        if exe.ends_with("target/debug") {
```

Comment:
True. (Only for when this logic is applicable: building Kani from the repo. So this shouldn't affect the typical user who will get things from the installer.)

Do you use release builds? I'm open to adding more logic to detect release too but then there should also be something to do the right thing when there's both builds available (the newest?)

Reply:
I think release build will be important for us when we add benchmarks.

What if we use a relative path to the current executable folder? This would keep it consistent with the cargo-kani build version.

Reply:
Sure. Can you please create an issue to fix scripts and the cargo-kani at the same time. They should all behave the same way.

Reply:
Sure https://github.com/model-checking/kani/issues/841